url20b0f880a5_jgraph_legacy_jgraph5_tgz-pJ8-org_jgraph_example_HelloWorldJ8/src/com/mxgraph/model/mxGeometry.java

```diff
@@ -0,0 +1,374 @@
+	 */
+	public mxGeometry()
+	{
+		this(0, 0, 0, 0);
+	}
+
+	/**
+	 * Constructs a geometry using the given parameters.
+	 * 
+	 * @param x X-coordinate of the new geometry.
+	 * @param y Y-coordinate of the new geometry.
+	 * @param width Width of the new geometry.
+	 * @param height Height of the new geometry.
+	 */
+	public mxGeometry(double x, double y, double width, double height)
+	{
+		super(x, y, width, height);
+	}
+
+	/**
+	 * Returns the alternate bounds.
+	 */
+	public mxRectangle getAlternateBounds()
+	{
+		return alternateBounds;
+	}
+
+	/**
+	 * Sets the alternate bounds to the given rectangle.
+	 * 
+	 * @param rect Rectangle to be used for the alternate bounds.
+	 */
+	public void setAlternateBounds(mxRectangle rect)
+	{
+		alternateBounds = rect;
+	}
+
+	/**
+	 * Returns the source point.
+	 * 
+	 * @return Returns the source point.
+	 */
+	public mxPoint getSourcePoint()
+	{
+		return sourcePoint;
+	}
+
+	/**
+	 * Sets the source point.
+	 * 
+	 * @param sourcePoint Source point to be used.
+	 */
+	public void setSourcePoint(mxPoint sourcePoint)
+	{
+		this.sourcePoint = sourcePoint;
+	}
+
+	/**
+	 * Returns the target point.
+	 * 
+	 * @return Returns the target point.
+	 */
+	public mxPoint getTargetPoint()
+	{
+		return targetPoint;
+	}
+
+	/**
+	 * Sets the target point.
+	 * 
+	 * @param targetPoint Target point to be used.
+	 */
+	public void setTargetPoint(mxPoint targetPoint)
+	{
+		this.targetPoint = targetPoint;
+	}
+
+	/**
+	 * Returns the list of control points.
+	 */
+	public List<mxPoint> getPoints()
+	{
+		return points;
+	}
+
+	/**
+	 * Sets the list of control points to the given list.
+	 * 
+	 * @param value List that contains the new control points.
+	 */
+	public void setPoints(List<mxPoint> value)
+	{
+		points = value;
+	}
+
+	/**
+	 * Returns the offset.
+	 */
+	public mxPoint getOffset()
+	{
+		return offset;
+	}
+
+	/**
+	 * Sets the offset to the given point.
+	 * 
+	 * @param offset Point to be used for the offset.
+	 */
+	public void setOffset(mxPoint offset)
+	{
+		this.offset = offset;
+	}
+
+	/**
+	 * Returns true of the geometry is relative.
+	 */
+	public boolean isRelative()
+	{
+		return relative;
+	}
+
+	/**
+	 * Sets the relative state of the geometry.
+	 * 
+	 * @param value Boolean value to be used as the new relative state.
+	 */
+	public void setRelative(boolean value)
+	{
+		relative = value;
+	}
+
+	/**
+	 * Swaps the x, y, width and height with the values stored in
+	 * alternateBounds and puts the previous values into alternateBounds as
+	 * a rectangle. This operation is carried-out in-place, that is, using the
+	 * existing geometry instance. If this operation is called during a graph
+	 * model transactional change, then the geometry should be cloned before
+	 * calling this method and setting the geometry of the cell using
+	 * mxGraphModel.setGeometry.
+	 */
+	public void swap()
+	{
+		if (alternateBounds != null)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It seems unusual to set the value of a variable only when it is already set. You might want to switch the null check of != on line 220 to == check before setting the field value on line 230 or verify if the check is needed at all.